code/modules/projectiles/guns/medbeam.dm

```diff
@@ -124,3 +124,201 @@
+	. = ..()
+
+	if(broken == INSTALL_CELL)
+		if(istype(attacking_item, /obj/item/stock_parts/cell/medbeam))
+			var/obj/item/stock_parts/cell/medbeam/battery = attacking_item
+			if(battery.charge != battery.maxcharge)
+				to_chat(user, "<span class='warning'>[src] won't function without a fully charged [battery].</span>")
+				return
+			to_chat(user, "<span class='notice'>You start replacing [src]'s battery.</span>")
+			attempt_repair(user, attacking_item, MULTITOOL_ELECTRONICS)
+			return
+
+	if(broken == INSTALL_ELECTRONICS)
+		if(istype(attacking_item, /obj/item/stack/cable_coil))
+			to_chat(user, "<span class='notice'>You start replacing the fried electronics in [src].</span>")
+			attempt_repair(user, attacking_item, INSTALL_CELL)
+			return
+
+	if(broken == INSTALL_LENS)
+		if(istype(attacking_item, /obj/item/stack/sheet/glass))
+			to_chat(user, "<span class='notice'>You start replacing the broken lens in [src].</span>")
+			attempt_repair(user, attacking_item, INSTALL_ELECTRONICS)
+			return
+
+/obj/item/gun/medbeam/damaged/screwdriver_act(mob/user, obj/item/screwdriver)
+	. = ..()
+	if(broken == SCREWDRIVER_OPEN)
+		to_chat(user, "<span class='notice'>You start removing the screws from [src]'s shell.</span>")
+		attempt_repair(user, screwdriver, REMOVE_OLD_PARTS)
+		return TRUE
+
+	if(broken == SCREWDRIVER_CLOSED)
+		to_chat(user, "<span class='notice'>You start replacing the screws on [src]'s shell.</span>")
+		attempt_repair(user, screwdriver, FALSE)
+		return TRUE
+
+/obj/item/gun/medbeam/damaged/crowbar_act(mob/living/user, obj/item/crowbar)
+	. = ..()
+	if(broken == REMOVE_OLD_PARTS)
+		to_chat(user, "<span class='notice'>You start prying out the old electronics and lens from [src].</span>")
+		attempt_repair(user, crowbar, WELD_SHELL)
+		return TRUE
+
+/obj/item/gun/medbeam/damaged/welder_act(mob/living/user, obj/item/welder)
+	. = ..()
+	if(broken == WELD_SHELL)
+		if(!tool_start_check(welder, user, 1))
+			to_chat(user, "<span class='warning'>[welder] isn't functioning.</span>")
+			return
+		to_chat(user, "<span class='notice'>You start welding [src] back to form.</span>")
+		attempt_repair(user, welder, INSTALL_LENS)
+		return TRUE
+
+/obj/item/gun/medbeam/damaged/multitool_act(mob/living/user, obj/item/multitool)
+	. = ..()
+	if(broken == MULTITOOL_ELECTRONICS)
+		if(user.electrocute_act(rand(1, 10), src))
+			do_sparks(6, FALSE, user)
+			return TRUE
+		to_chat(user, "<span class='notice'>You start to activate the electronics in [src].</span>")
+		attempt_repair(user, multitool, SCREWDRIVER_CLOSED)
+		return TRUE
+
+/obj/item/gun/medbeam/damaged/proc/attempt_repair(mob/living/user, obj/item/tool_used, next_broken_state)
+	if(!do_after(user, 5 SECONDS, TRUE, src))
+		to_chat(user, "<span class='notice'>You stop repairing [src].</span>")
+		return
+
+	if(!broken)
+		to_chat(user, "<span class='warning'>[src] isn't broken, what are you trying to repair?</span>")
+		return
```

Comment:
is it expected that this proc will be called when `broken` is FALSE/0/null or is this just a contingency check? If not I would just return here